packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -129,10 +138,13 @@ export default class OperatorModeStateService extends Service {
   }
 
   shiftStack(stack: StackItem[], destinationIndex: number) {
-    stack.forEach((item) => {
-      this.trimItemsFromStack(item);
+    let stackItemsCopy = stack.slice(); // The actions in the loop are mutating the stack items, so we need to make a copy to make sure to iterate over all items from the original stack
```

Comment:
make sense i had to do something similar for the auto save. perhaps a clearer way to writing this is to:
```js
let stackItemsCopy = [...stack]
```
I was a little confused why we were using slice at first since the desire is to copy the array not trim it, slice just happens to have this side effect.

Reply:
Agree, it is clearer –`slice()` is confusing for this case. Will change it to what you suggested